tests/e2e/core-tests/specs/merchant/wp-admin-order-apply-coupon.test.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable jest/no-export */
+
+/**
+ * Internal dependencies
+ */
+const {
+	StoreOwnerFlow,
+	createSimpleProduct,
+	createSimpleOrder,
+	createCoupon,
+	uiUnblocked,
+	addProductToOrder,
+} = require( '@woocommerce/e2e-utils' );
+
+const config = require( 'config' );
+const simpleProductName = config.get( 'products.simple.name' );
+
+const couponDialogMessage = 'Enter a coupon code to apply. Discounts are applied to line totals, before taxes.';
+
+let couponCode;
+let orderId;
+
+const runOrderApplyCouponTest = () => {
+	describe('WooCommerce Orders > Apply coupon', () => {
+		beforeAll(async () => {
+			await StoreOwnerFlow.login();
+			await Promise.all([
+				await createSimpleProduct(),
+				couponCode = await createCoupon(),
+				orderId = await createSimpleOrder('Pending payment', simpleProductName),
+				await addProductToOrder(orderId, simpleProductName),
+
+				// We need to remove any listeners on the `dialog` event otherwise we can't catch the dialog below
+				page.removeAllListeners('dialog'),
+			]);
+		} );
+
+		it('can apply a coupon', async () => {
+			const couponDialog = await expect(page).toDisplayDialog(async () => {
+				await expect(page).toClick('button.add-coupon');
+			});
+
+			expect(couponDialog.message()).toMatch(couponDialogMessage);
+
+			// Accept the dialog with the coupon code
+			await couponDialog.accept(couponCode);
+
+			await uiUnblocked();
+
+			// Verify the coupon list is showing
+			await page.waitForSelector('.wc-used-coupons');
+			await expect(page).toMatchElement('.wc_coupon_list', { text: 'Coupon(s)' });
+			await expect(page).toMatchElement('.wc_coupon_list li.code.editable', { text: couponCode });
+
+			// Check that the coupon has been applied
+			await expect(page).toMatchElement('.wc-order-item-discount', { text: '5.00' });
+			await expect(page).toMatchElement('.line_cost > .view > .woocommerce-Price-amount', { text: '4.99' });
```

Comment:
I think this is fine for this PR but maybe we can do something similar (or as part of it)to https://github.com/woocommerce/woocommerce/pull/28575 for these card coded values later.

Reply:
Agreed! I was thinking I would circle back on that PR and update these tests to use the new config variable once they're in `master`.